### PR TITLE
> fix Waypoint NACK when there are no pods

### LINKED ISSUE #43266: Waypoint NACK when there are no pods
```
2023-02-10T01:00:47.734985Z     warning envoy config external/envoy/source/common/config/grpc_subscription_impl.cc:128  gRPC config for type.googleapis.com/envoy.config.listener.v3.Listener rejected: Error adding/updating listener(s) internal: Proto constraint validation failed (IPMatcherValidationError.RangeMatchers[0]: embedded message failed validation | caused by IPRangeMatcherValidationError.Ranges: value must contain at least 1 item(s)): range_matchers {
  on_match {
    action {
      name: "direct"
      typed_config {
        [type.googleapis.com/google.protobuf.StringValue] {
          value: "direct"
        }
      }
    }
  }
}

```

Should be a simple fix

### PR DESCRIPTION
**Pull request by kebe7jun**
**Please provide a description of this PR:**

Fix https://github.com/istio/istio/issues/43266

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/listener_waypoint.go [hunk 1]
```diff
@@ -294,12 +294,14 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []WorkloadAndServices, svcs
 				PrefixLen:     cidr.PrefixLen,
 			})
 		}
-		chains = append(chains, directChain)
-		ipMatcher.RangeMatchers = append(ipMatcher.RangeMatchers,
-			&matcher.IPMatcher_IPRangeMatcher{
-				Ranges:  ipRange,
-				OnMatch: match.ToChain(directChain.Name),
-			})
+		if len(ipRange) > 0 {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/listener_waypoint.go:297]: can check len(wls) in L275

**Review by hzxuzhonghu — COMMENTED** [review]
lgtm

File: pilot/pkg/networking/core/v1alpha3/listener_waypoint.go [hunk 2]
```diff
@@ -271,7 +271,7 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []WorkloadAndServices, svcs
 		}
 	}
 
-	{
+	if len(wls) > 0 {
 		// Direct pod access chain.
 		directChain := &listener.FilterChain{
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/listener_waypoint.go:276]: can we keep the chain  and just never match to it

Comment (kyessenov) [reply pilot/pkg/networking/core/v1alpha3/listener_waypoint.go:276]: This is not a right fix. I think it's CIDR validation that gets triggered on empty set.

**Comment by howardjohn** [comment]
Can you rebase this on master now that we have upstreamed? You can do so in the UI by clicking "edit"  at the top

**Review by howardjohn — APPROVED** [review]

**Review by jacob-delgado — APPROVED** [review]

**Pull request merged**